lib/ansible/modules/cloud/azure/azure_rm_galleryimageversion.py

```diff
@@ -363,9 +365,13 @@ def exec_module(self, **kwargs):
                 if not self.default_compare(modifiers, self.body, old_response, '', self.results):
                     self.to_do = Actions.Update
 
-        # fix leftovers (if empty structures were left)
-        self.body.get('properties', {}).get('publishingProfile', {}).pop('snapshot', None)
-        self.body.get('properties', {}).get('publishingProfile', {}).pop('managed_image', None)
+        # fix for differences between version 2019-03-01 and 2019-07-01
+        snapshot = self.body.get('properties', {}).get('publishingProfile', {}).pop('snapshot', None)
+        if snapshot is not None:
+            self.body['properties'].setdefault('storageProfile', {}).setdefault('osDiskImage', {}).setdefault('source', {})['id'] = snapshot
+        managed_image = self.body.get('properties', {}).get('publishingProfile', {}).pop('managed_image', None)
+        if managed_image:
```

Comment:
@zikalino Just a quick question: is there a reason to check for None for snapshot but just check truthiness for managed_image?  The difference, coming right after each other is very noticable but there's no comment to indicate why that might have been done.  If there is a reason, a comment is probably appropriate.  If there isn't a reason, making them both the same will give future maintainers one less thing to worry about when they change it ;-)

This comment doesn't block the PR.  I'll go ahead and merge the backport now.